optimade/server/middleware.py

```diff
@@ -387,7 +386,7 @@ def showwarning(
                     # When a warning is logged during Python shutdown, linecache
                     # and the import machinery don't work anymore
                     line = None
-                    linecache = None
+                    linecache = None  # type: ignore[assignment]
```

Comment:
Why do you set `linecache` to None here?  It does not seem to be used afterwords.
```suggestion
```

Reply:
The middleware warnings stuff is a bit fiddly and requires a few hacks like this. `linecache` is the name of a module that is otherwise not imported (see the comment section).

We shouldn't change this without good reason, hence why I just added the ignore

Reply:
In the function you took this code from `linecache is still used later on in the function. So it makes sense to set it to `None` In our case we do not use the value of `linecache` later on so this kine can be removed.
In fact we could also replace the line above (`line = None`) with pass as the value of line is already `None`.

Reply:
Probably, but I don't think I want to do that in this PR. I didn't write this stuff initially (assume it was Casper) and there is limited benefit to changing it. The fact we are using private interfaces with the Python warnings library and linecache itself should indicate that this is a hacky bit of code!